### PR TITLE
feat: add type to MAE payload && update docs

### PR DESCRIPTION
**Pull request by JiaoMaWHU**
## Summary
1. add a `MetadataEventType` generic enum with three value.
2. include this enum in MAE payload
3. update doc with https://github.com/linkedin/datahub-gma/pull/299

## Testing
./gradlew build

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl [hunk 1]
```diff
@@ -0,0 +1,23 @@
+namespace com.linkedin.metadata.events
```
Comment (jsdonn) [inline dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl:1]: are we adding this to both MCE and MAE? will you add to MCE in a later PR?

**Review by jsdonn — CHANGES_REQUESTED** [review]

File: dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl (hunk 1 above)
Comment (JiaoMaWHU) [reply dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl:1]: I will not add it into MCE now. But the name `MetadataEventType` is trying to accommodate such need. We can reuse this payload if MCE needs it in the future.

File: dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Enum that represents the type of a metadata event. The metadata event consumers decide processing strategy based
+ * on this type.
+ */
+enum MetadataEventType {
+
+  /**
+   * Default type. Standard processing strategy applies.
```
[the remark this answers is not in the document]
Comment (JiaoMaWHU) [reply dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl:10]: Sure, that's a good suggestion.

File: dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+   */
+  DEFAULT
+
+  /**
+   * Backfill events are used to backfill metadata that are skipped or failed in previous processing.
+   */
+  BACKILL
```
[the remark this answers is not in the document]
Comment (JiaoMaWHU) [reply dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl:17]: will update

File: dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  /**
+   * Indicates this event is to bootstrap a new type of metadata.
```
[the remark this answers is not in the document]
Comment (JiaoMaWHU) [reply dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl:20]: Yes. But your comment made me realize BOOTSTRAP is not limited to new metadata onboarding, I will update the doc.

File: dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl (hunk 2 above)
[the remark this answers is not in the document]
Comment (yangyangv2) [reply dao-api/src/main/pegasus/com/linkedin/metadata/events/MetadataEventType.pdl:10]: How about MetadataIngestionMode?

**Review by jsdonn — APPROVED** [review]
lgtm

**Pull request merged**